### PR TITLE
Dry up AcknowledgedResponse Handling

### PR DESCRIPTION
**Pull request by original-brownbear**
Motivated by the fact that we keep adding more and more master node actions that return an ack response:

1. `AcknowledgedResponse` should really be two singletons to make things clearer and save some objects and code size.
2. We were duplicating reading this type over and over  in a bunch of transport master node actions so I dried that up

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/support/master/AcknowledgedTransportMasterNodeAction.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+
+public abstract class AcknowledgedTransportMasterNodeAction<Request extends MasterNodeRequest<Request>>
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/action/support/master/AcknowledgedTransportMasterNodeAction.java:32]: short javadoc?

**Review by rjernst — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/action/support/master/AcknowledgedTransportMasterNodeAction.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/action/support/master/AcknowledgedTransportMasterNodeAction.java:35]: added :)

**Comment by original-brownbear** [comment]
Thanks Ryan!

**Pull request merged**